packages/components/src/internal/components/entities/utils.ts

```diff
@@ -155,6 +155,49 @@ export function getUniqueIdColumnMetadata(queryInfo: QueryInfo): Map<string, Edi
     return columnMetadata;
 }
 
+export function getEntityNoun(entityDataType: EntityDataType, quantity: number): string {
+    return quantity === 1 ? entityDataType.nounSingular : entityDataType.nounPlural;
+}
+
 export function getEntityDescription(entityDataType: EntityDataType, quantity: number): string {
     return quantity === 1 ? entityDataType.descriptionSingular : entityDataType.descriptionPlural;
 }
+
+export function getUpdatedLineageRowsForBulkEdit(nonAliquots: Record<string, any>,
+                                                 selectedParents: List<EntityChoice>,
+                                                 originalParents: Record<string, List<EntityChoice>>,
+                                                 queryInfo: QueryInfo) : any[] {
+    const rows = [];
+    Object.keys(nonAliquots).forEach((rowId) => {
+        let updatedValues = {}
+        let haveUpdate = false;
+
+        // Find the types that are included and use those for change comparison.
+        // Types that are not represented in the selected parents won't be changed.
+        selectedParents.forEach(selected => {
+            let originalValue = null;
+            const possibleChange = originalParents[rowId].find(p => p.type.lsid == selected.type.lsid);
+            if (possibleChange) {
+                originalValue = possibleChange.gridValues.map(gridValue => gridValue.displayValue).sort(naturalSort).join(",");
+            }
+            const selValue = selected.value ? selected.value.split(",").sort(naturalSort).join(",") : null;
```

Comment:
nice, you did decide to do the "split, sort, join". do you think I should do the same for my comparison?